nightfall-optimist/src/routes/proposer.mjs

```diff
@@ -67,14 +69,22 @@ router.post('/register', async (req, res, next) => {
         logger.warn(
           'Proposer was already registered on the blockchain but not with this Optimist instance - registering locally',
         );
-        const stateContractInstance = await waitForContract(STATE_CONTRACT_NAME);
-        const currentProposer = await stateContractInstance.methods.getCurrentProposer().call();
+        stateContractInstance = await waitForContract(STATE_CONTRACT_NAME);
+        currentProposer = await stateContractInstance.methods.getCurrentProposer().call();
```

Comment:
Hey @druiz0992, why do not reuse the same instances from the lines 60-61?

Reply:
You’re right, thanks